nucleus/v1/types.go

```diff
@@ -0,0 +1,187 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+
+// NucleusHub represents a deployment of nucleus hub component.
+type NucleusHub struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec represents a desired deployment configuration of nucleus hub
+	Spec NucleusHubSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+
+	// Status represents the current status of nucleus hub
+	// +optional
+	Status NucleusHubStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// NucleusHubSpec represents a desired deployment configuration of nucleus hub.
+type NucleusHubSpec struct {
+	// Components represents the components to be installed for nucleus hub.
+	// +required
+	Components NucleusHubComponents `json:"components" protobuf:"bytes,1,opt,name=components"`
```

Comment:
We should not imply that there are components.  I don't think this infrastructure should use component style language or people will think that they add components.  I don't think we will add any components.  Instead of components, we need to make the deployment across spokes really hum using a local OLM to manage installed components and provide a solid pattern for other operators to use to provide hub/spoke mechanics if they need them.

Reply:
+1 on avoiding characterization as components. The API here should be _just_ what is necessary to deploy the most essential pieces of the system, which currently would be `registration` and `work`.

I think you can just inline `Registration` and add `Work` as a sibling.